src/main/java/org/jenkinsci/account/Application.java

```diff
@@ -254,6 +242,22 @@ public HttpResponse doDoSignup(
             }
         }
 
+        final DirContext con = connect();
+        try {
+            if(ldapObjectExists(con, "(id={0})", userid)) {
+                throw new UserError("ID " + userid + " is already taken. Perhaps you already have an account imported from legacy java.net? You may try resetting the password.");
```

Comment:
I'm not sure if there's an attack vector here by confirming the presence of emails or user IDs in the system. I'm not thinking of account creation, so much as compromising existing accounts.